source/dub/dependency.d

```diff
@@ -33,9 +33,9 @@ struct Dependency {
 	private {
 		// Shortcut to create >=0.0.0
 		enum ANY_IDENT = "*";
-		string m_cmpA;
+		bool m_inclusiveA = true; // A comparison > (true) or >= (false)
 		Version m_versA;
-		string m_cmpB;
+		bool m_inclusiveB = true; // B comparison < (true) or <= (false)
```

Comment:
"([" is definitely a convenient solution for template interfaces, but for the internal representation it seems to be sub-optimal. It means having magic constants in the code and using a type that has a greater value range than the problem domain.

Maybe a dedicated interval type, usable at runtime or at compile time, in Phobos would be nice: `Interval!int(3, 15, "(]")`
This would then define common operations (union, intersection, membership test and possibly arithmetic operations on the limits) and could then internally drive all places that work with such ranges.

BTW, I'll rename the `A` and `B` to `Min` and `Max` to make their semantics clearer.